### ISSUE TITLE
Open chat in Hexchat automatically ?

### ISSUE THREAD

**Issue by TheLongLife** [streamlink/streamlink-twitch-gui#406]
Can someone tell me if I can set that up and if yes, how ?

**Comment by bastimeyer** [comment]
**TL;DR:**
Use Chatty

-----

I'm not sure if it's possible in a nice way, since there are a couple of issues.

First off, not all channels can be accessed by connecting to the same Twitch irc server. Usually, you need to query the Twitch API first in order to retrieve the correct server address and port. This depends on the size of the channel, but smaller channels are always accessible by connecting to `irc.twitch.tv:6667`. For larger channels, there are also dedicated servers for the JOIN and PART messages, but this shouldn't be of interest here.

The main issue with Hexchat is that it doesn't have the necessary command line parameters which [Chatty has](https://github.com/streamlink/streamlink-twitch-gui/blob/v1.1.1/src/config/chat.json#L67), so you'll need to write a wrapper shell script for launching hexchat. Then you simply need to know the necessary command line parameters of hexchat (`man hexchat`) and some hexchat client commands (`/help -l` in Hexchat).

***hexchat-twitch.sh:***
```
#!/usr/bin/env bash

hexchat -e -c "NICK ${1}" &
sleep 1
hexchat -e -c "SERVER irc.twitch.tv 6667 oauth:${2}" &
sleep 3
hexchat -e -c "JOIN #${3}" &
```

**Custom "command line" in the chat settings menu:**
```
/path/to/hexchat-twitch.sh "{user}" "{token}" "{channel}"
```

The `-e`/`--existing` parameter is only available on Linux (requires DBus), so you won't be able to run this on MacOS or on Windows (if you translate this bash script to powershell). Since hexchat doesn't wait for commands to be executed and since there's also no way of knowing when hexchat has successfully connected to the irc server, all commands have to be executed one after each other with a sleep command in between (hexchat/hexchat#884).

Please don't ask me how to set this up when using hexchat with multiple, different IRC servers.

-----

See this link for a general IRC client tutorial:
https://help.twitch.tv/customer/portal/articles/1302780-twitch-irc

**Comment by TheLongLife** [comment]
Ok, thanks for the info and explanation. I guess I'll use Chatty than since I'm on Windows.
